pkg/quota/generic/evaluator.go

```diff
@@ -69,6 +69,9 @@ type ListFuncByNamespace func(namespace string) ([]runtime.Object, error)
 // MatchesScopeFunc knows how to evaluate if an object matches a scope
 type MatchesScopeFunc func(scope api.ResourceQuotaScope, object runtime.Object) (bool, error)
 
+// MatchesPriorityClassScopeFunc knows how to evaluate if an object matches a scope
```

Comment:
why do we need a new scope func type?

this is a scope like any other scope and should be handled by modifying the MatchesScopeFunc already.